src/mavoscript.js

```diff
@@ -321,7 +321,62 @@ var _ = Mavo.Script = {
 				return ret;
 			},
 			precedence: 3
-		}
+		},
+		"groupby": {
+			symbol: "by",
+			code: (array, key) => {
+				if (Array.isArray(array) && Array.isArray(key))  { 
+					var propName = key[Mavo.as] === true ? key[Mavo.name] : $.value(key[0], Mavo.toNode, "property");
+					var temp = new Mavo.BucketMap({arrays: true});
+					var ret = [];
+					ret[Mavo.groupedBy] = true;
+
+					for (var i = 0; i < array.length; i++) {
+						const k = i < key.length ? Mavo.value(key[i]) : null;
+						temp.set(k, Mavo.value(array[i]));
+					}	
+
+					for (var [value, items] of temp) {
```

Comment:
Avoid using `for..of` loops for now. They are lovely, but when transpiled by Babel they produce a bloated mess. See: http://babeljs.io/repl#?babili=false&browsers=last%204%20versions%2C%20Chrome%20%3E%3D%2041%2C%20IE%2011&build=&builtIns=false&spec=false&loose=false&code_lz=GYewTgBAFAbghpAlhEwILASggbwFAQR4C-QA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&sourceType=module&lineWrap=false&presets=stage-2%2Cenv&prettier=false&targets=&version=6.26.0&envVersion=1.6.2

Use `array.forEach` instead.